services/core/shared/core/compat/sdk_v5/snapshotUtils.js

```diff
@@ -0,0 +1,112 @@
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const path = require('path');
+const util = require('util');
+const execInShell = util.promisify(require('child_process').exec);
+
+const {
+	CacheRedis,
+	Logger,
+	Exceptions: { NotFoundException },
+} = require('lisk-service-framework');
+
+const {
+	exists,
+	mkdir,
+} = require('../../../fsUtils');
+
+const config = require('../../../../config');
+const {
+	downloadAndUnzipFile,
+} = require('../../../downloadFile');
+
+const logger = Logger();
+
+let snapshotUrl;
+let snapshotFilePath;
+
+const constantsCache = CacheRedis('networkConstants', config.endpoints.redis);
+
+const downloadSnapshot = async () => {
+	const directoryPath = path.dirname(snapshotFilePath);
+	if (!(await exists(directoryPath))) await mkdir(directoryPath, { recursive: true });
+	await downloadAndUnzipFile(snapshotUrl, snapshotFilePath);
+};
+
+const checkCommandAvailability = async () => {
+	const { stdout: mysqlAvailable } = await execInShell('which mysql');
+	const { stdout: dockerComposeAvailable } = await execInShell('which docker-compose');
+	if (!mysqlAvailable && !dockerComposeAvailable) throw new NotFoundException('Both mysql and docker-compose are unavailable in PATH');
+};
+
+const resolveSnapshotRestoreCommand = async (connEndpoint) => {
+	await checkCommandAvailability();
+	const composeFile = config.snapshot.composeFilePath;
+	const dockerServiceName = config.snapshot.serviceName;
+
+	const [user, password] = connEndpoint.split('//')[1].split('@')[0].split(':');
+	const [host, port, database] = connEndpoint.split('@')[1].split(new RegExp('/|:', 'g'));
+	const mysqlSnapshotCommand = `mysql ${database} -h ${host} -P ${port} -u ${user} -p${password} < ${snapshotFilePath}`;
+	if (!composeFile && !dockerServiceName) return mysqlSnapshotCommand;
+
+	if (composeFile) {
+		if (!dockerServiceName) throw new Error('Docker service name required');
+		return `docker-compose -f ${composeFile} exec -T ${dockerServiceName} ${mysqlSnapshotCommand}`;
+	}
+	throw new Error('Snapshot restore command cannot be resolved');
+};
+
+const applySnapshot = async (connEndpoint = config.endpoints.mysql) => {
+	try {
+		logger.debug('Attempting to resolve the snapshot command');
+		const snapshotRestoreCommand = await resolveSnapshotRestoreCommand(connEndpoint);
+		logger.debug(`Resolved snapshot command to: ${snapshotRestoreCommand}`);
+		logger.info('Attempting to apply snapshot');
+		const { stdout, stderr } = await execInShell(snapshotRestoreCommand);
+		logger.info(stdout);
+		logger.warn(stderr);
+		logger.info('SQL file(s) imported successfully');
+	} catch (error) {
+		logger.error(error);
+		throw error;
+	}
+};
+
+const initSnapshot = async () => {
+	if (config.snapshot.enable !== true) {
+		logger.info('Index snapshot application has been disabled');
+		return;
+	}
+
+	const { data: { networkIdentifier } } = JSON.parse(await constantsCache.get('networkConstants'));
```

Comment:
Please make the following changes to handle the cold start scenario.
```suggestion
	const cachedNetworkConstants = await constantsCache.get('networkConstants');
	const {
		data: { networkIdentifier },
	} = cachedNetworkConstants ? JSON.parse(cachedNetworkConstants) : await getNetworkConstants();
```